src/state_history_pg.hpp

```diff
@@ -145,48 +150,75 @@ inline std::string sql_str(bool bulk, uint32_t v)
-inline std::string sql_str(bool bulk, abieos::varuint32 v)                              { return std::string(v); }
-inline std::string sql_str(bool bulk, abieos::varint32 v)                               { return std::string(v); }
-inline std::string sql_str(bool bulk, const abieos::int128& v)                          { return std::string(v); }
-inline std::string sql_str(bool bulk, const abieos::uint128& v)                         { return std::string(v); }
-inline std::string sql_str(bool bulk, const abieos::float128& v)                        { return quote_bytea(bulk, std::string(v)); }
-inline std::string sql_str(bool bulk, abieos::name v)                                   { return quote(bulk, v.value ? std::string(v) : std::string()); }
-inline std::string sql_str(bool bulk, abieos::time_point v)                             { return v.microseconds ? quote(bulk, std::string(v)): null_value(bulk); }
-inline std::string sql_str(bool bulk, abieos::time_point_sec v)                         { return v.utc_seconds ? quote(bulk, std::string(v)): null_value(bulk); }
-inline std::string sql_str(bool bulk, abieos::block_timestamp v)                        { return v.slot ?  quote(bulk, std::string(v)) : null_value(bulk); }
-inline std::string sql_str(bool bulk, const abieos::checksum256& v)                     { return quote(bulk, v.value == abieos::checksum256{}.value ? "" : std::string(v)); }
-inline std::string sql_str(bool bulk, const abieos::public_key& v)                      { return quote(bulk, public_key_to_string(v)); }
-inline std::string sql_str(bool bulk, const abieos::signature& v)                       { return quote(bulk, signature_to_string(v)); }
-inline std::string sql_str(bool bulk, const abieos::bytes&)                             { throw std::runtime_error("sql_str(bytes): not implemented"); }
-inline std::string sql_str(bool bulk, transaction_status v)                             { return quote(bulk, to_string(v)); }
-inline std::string sql_str(bool bulk, abieos::symbol v)                                 { return quote(bulk, abieos::symbol_to_string(v.value)); }
+inline std::string sql_str(bool bulk, eosio::varuint32 v)                               { return std::to_string(v.value); }
+inline std::string sql_str(bool bulk, eosio::varint32 v)                                { return std::to_string(v.value); }
+#ifndef ABIEOS_NO_INT128
+inline std::string sql_str(bool bulk, const abieos::int128& v)                          { auto nv = -v; return v < 0 ? std::string("-") + abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(nv)).get_array()) : abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
+inline std::string sql_str(bool bulk, const abieos::uint128& v)                         { return abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
```

Comment:
abieos's `uint128` and `int128` structs define a cast to `std::string`

Reply:
In Clang on Ubuntu, `__int128` is available but corresponding standard library functions to convert it don't seem to be.  Current CMake behavior does not define `ABIEOS_NO_INT128`, so `struct uint128` and `struct int128` are not available.  Should that be changed?  Neither of them in abieos.hpp provides a `std::string` operator.

Reply:
Got it. In that case, you might want to create separate conversion functions for those. Also, that `reinterpret_cast` is a strict-aliasing violation for both the `__int128` and `struct int128` cases.

Reply:
Added converters for non-native integers.